app/controllers/reports/patient_lists_controller.rb

```diff
@@ -1,11 +1,41 @@
 class Reports::PatientListsController < AdminController
+  include ZipTricks::RailsStreaming
+
   attr_reader :scope, :region, :download_params
 
   before_action :set_scope, only: [:show, :diabetes]
   before_action :set_region, only: [:show, :diabetes]
   before_action :set_download_params, only: [:show, :diabetes]
 
   def show
+    if Flipper.enabled?(:patient_list_direct_download)
+      Datadog::Tracing.trace("Direct patient line list download") do
+        patients = @region.assigned_patients.excluding_dead
+        enumerator = PatientsWithHistoryExporter.csv_enumerator(patients)
+
+        file_name = "patient-list_#{@region.region_type}_#{@region.name}_#{I18n.l(Date.current)}"
+        csv_file_name = "#{file_name}.csv"
+        zip_archive_name = "#{file_name}.zip"
+        response.headers["Content-Disposition"] = ActionDispatch::Http::ContentDisposition.format(
+          disposition: "attachment",
+          filename: zip_archive_name
+        )
+
+        first_row = enumerator.peek
+        response.headers["Last-Modified"] = first_row.second
+
+        return zip_tricks_stream do |zip|
+          zip.write_deflated_file(csv_file_name) do |sink|
+            CSV(sink) do |csv_write|
+              enumerator.each do |row|
+                csv_write << row
+              end
+            end
+          end
+        end
```

Comment:
[minor, nonblocking] This part leaks CSV generation logic into the controller.

For an enumerator created with the `#csv_enumerator` method, I'd have expected an object that yields actual CSV strings, but instead it yields plain arrays—this is why we need to now call the `CSV` module in our controller to actually perform the conversion. Moreover, it leads to 3 layers of block indentation which is a little tricky to parse.

As an alternate proposal, I have annotated a suggestion in the `patients_with_history_exporter.rb` that would make this enumerator return strings instead of arrays. Once that happens, we could rewrite this section as:

```suggestion
        return zip_tricks_stream do |zip|
          zip.write_deflated_file(csv_file_name) do |sink|
              enumerator.each { |row| sink << row }
          end
        end
```

FYI, I'm okay with the current code also and I haven't fully tested if my above proposal works. Consider this suggestion nonblocking.